solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/GabrielaSillis/GSLoginViewController.swift

```diff
@@ -223,3 +176,84 @@ extension GSLoginViewController {
         loginButton.isEnabled = true
     }
 }
+
+private extension GSLoginViewController {
+    func configureTextFieldWithDefaultValue(
+        emailTextField: String = "clean.code@devpass.com",
+        passwordTextField: String = "111111") {
+        self.emailTextField.text = emailTextField
+        self.passwordTextField.text = passwordTextField
+    }
+    
+    func verifyIfUserIsAlreadyLoggedIn() {
+        if let _ = UserDefaultsManager.UserInfos.shared.readSesion() {
+            coordinator.startNavigatingFlow()
+        }
+    }
+    
+    func checkDeviceConnectivityAnduRequestAuthentication() {
+        let deviceDoNotHaveConnectivity = !ConnectivityManager.shared.isConnected
+        if deviceDoNotHaveConnectivity {
```

Comment:
`isDeviceConnectedToInternet`
em if é sempre interessante começar com o is, pq se assemelha melhor com a lingua ingles

Reply:
boa, acabei esquecendo disso, renomeie

Reply:
@ppedroam uma dúvida que apareceu aqui, é se essa verificação poderia estar na viewModel?